articles/rules/preview/index.md

```diff
@@ -0,0 +1,314 @@
+---
+title: Rules
+description: Learn what Rules are and how you can use them to customize and extend Auth0's capabilities.
+toc: true
+---
+
+# Rules
+
+<div class="alert alert-info">
+<strong>Heads up!</strong> This article describes the latest Auth0 implementation, according to the <a href="/api-auth">API Authorization flows</a>. If you are looking for the legacy Rules doc refer to <a href="/rules">Rules</a>.
+</div>
+
+**Rules** are functions written in JavaScript that are executed in Auth0 as part of the transaction every time a user authenticates to your application. They are executed after the authentication and before the authorization.
+
+Rules allow you to easily customize and extend Auth0's capabilities. They can be chained together for modular coding and can be turned on and off individually.
+
+![](/media/articles/rules/flow.png)
+
+* **Step 1:** An app initiates an authentication request to Auth0.
+* **Step 2:** Auth0 routes the request to an Identity Provider through a configured connection.
+* **Step 3:** The user authenticates successfully.
+* **Step 4:** The `user` object representing the logged in user is passed through the Rules pipeline, and returned to the app.
```

Comment:
in the new flows, we don't return the user object to the app. we return any of {id_token, token, code}.